using-mobile-extensions/adobe-experience-platform-messaging/messaging-api-reference.md

```diff
@@ -0,0 +1,420 @@
+# Messaging API reference
+
+## extensionVersion <a id="extensionversion"></a>
+
+Returns the version of the client-side Consent extension.
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+#### Syntax
+
+```java
+public static String extensionVersion();
+```
+
+#### Example
+
+```java
+Messaging.extensionVersion();
+```
+{% endtab %}
+
+{% tab title="iOS" %}
+### Swift
+
+#### Syntax
+
+```swift
+public static let extensionVersion
+```
+
+#### Example
+
+```swift
+Messaging.extensionVersion
+```
+
+### Objective-C
+
+#### Syntax
+
+```swift
+public static let extensionVersion
+```
+
+#### Example
+
+```objective-c
+[AEPMobileMessaging extensionVersion];
+```
+{% endtab %}
+{% endtabs %}
+
+## handleNotificationResponse<a id="handlenotificationresponse"></a>
+
+Sends the push notification interactions feedback to experience platform.
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+#### Syntax
+
+```java
+public static void handleNotificationResponse(final Intent intent, final boolean applicationOpened, final String customActionId);
+```
+
+* _intent_ - Intent which contains information related to messageId and data.
+* _applicationOpened_ - boolean values denoting whether the application was opened or not. 
+* _actionId_ - String values denoting the id of the custom action.
+
+#### Example
+
+```java
+Messaging.handleNotificationResponse(intent, true, "customActionId");
+```
+{% endtab %}
+
+{% tab title="iOS" %}
+### Swift
+
+#### Syntax
+
+```swift
+static func handleNotificationResponse(_ response: UNNotificationResponse, applicationOpened: Bool, customActionId: String?)
+```
+
+* _response_ - UNNotificationResponse response object containing push notification details 
+* _applicationOpened_ - Bool values denoting whether the application was opened or not. 
+* _customActionId_ - String values denoting the id of the custom action.
+
+#### Example
```

Comment:
for these examples, it would be good to provide some context around where the method is being called.  specifically, since you're passing through an `Intent` for android and a `UNNotificationResponse` in iOS, it's helpful to show from which methods these objects are being retrieved.